scripts/zones/Ceizak_Battlegrounds/mobs/Unfettered_Twitherym.lua

```diff
@@ -0,0 +1,110 @@
+    Dark = { xi.skillchainType.COMPRESSION, xi.skillchainType.DARKNESS, xi.skillchainType.DARKNESS_II },
+    Light = { xi.skillchainType.TRANSFIXION, xi.skillchainType.LIGHT, xi.skillchainType.LIGHT_II }
+}
+
+local chosenElement = nil
+local phaseDTApplied = {
+    ['74_49'] = false,
+    ['49_24'] = false,
+    ['24_0'] = false
+}
+
+local function chooseAffinity()
+    local affinities = { 'Fire', 'Water', 'Thunder', 'Wind', 'Ice', 'Earth', 'Dark', 'Light' }
+    chosenElement = affinities[math.random(#affinities)]
+end
+
+local function isSkillchainCorrect(skillchainID)
+    local affinitySkillchains = skillchainAffinities[chosenElement]
+    for _, chain in ipairs(affinitySkillchains) do
+        if skillchainID == chain then
+            return true
+        end
+    end
+    return false
+end
+
+local function applyDamageReduction(mob, phase)
+    if not phaseDTApplied[phase] then
+        mob:setMod(xi.mod.UDMGPHYS, damageReductionValue)
+        mob:setMod(xi.mod.UDMGMAGIC, damageReductionValue)
+        mob:setMod(xi.mod.UDMGRANGE, damageReductionValue)
+        mob:setMod(xi.mod.UDMGBREATH, damageReductionValue)
+        phaseDTApplied[phase] = true
+    end
+end
+
+local function resetDamageModifiers(mob)
+    for _, mod in ipairs({ xi.mod.UDMGPHYS, xi.mod.UDMGMAGIC, xi.mod.UDMGRANGE, xi.mod.UDMGBREATH }) do
+        mob:setMod(mod, originalDamage)
+    end
+end
+
+local function getCurrentPhase(mob)
+    local hpp = mob:getHPP()
+    if hpp <= 74 and hpp > 49 then
+        return '74_49'
+    elseif hpp <= 49 and hpp > 24 then
+        return '49_24'
+    elseif hpp <= 24 then
+        return '24_0'
+    end
+    return nil
+end
+
+entity.onMobFight = function(mob)
+    local phase = getCurrentPhase(mob)
+    if mob:getHPP() <= 50 then
+        mob:setMobMod(xi.mobMod.SKILL_LIST, 2018)
+    else
+        mob:setMobMod(xi.mobMod.SKILL_LIST, 2017)
+    end
+end
+
+entity.onMobSpawn = function(mob)
+    originalDamage = mob:getMod(xi.mod.UDMGPHYS)
+
+    mob:addListener('WEAPONSKILL_USE', 'ANY_MOBSKILL_CHECK', function(mob, Target, Skillid, TP, action)
+        local phase = getCurrentPhase(mob)
+        if phase and not phaseDTApplied[phase] then
+            applyDamageReduction(mob, phase)
+            if mob:getHP() <= mob:getMaxHP() * 0.75 then
+                chooseAffinity()
+            end
+        end
+    end)
+
+    mob:addListener('WEAPONSKILL_TAKE', 'SKILLCHAIN_DETECT', function(Target, mob, Skillid, TP, action)
+        if mob:hasStatusEffect(xi.effect.SKILLCHAIN) and chosenElement then
+            local skillchainEffect = mob:getStatusEffect(xi.effect.SKILLCHAIN)
+            local power = skillchainEffect:getPower()
+            if isSkillchainCorrect(power) then
+                resetDamageModifiers(mob)
+                chosenElement = nil
+            end
+        end
+    end)
+end
+
+entity.onMobDeath = function(mob)
+    mob:removeListener('ANY_MOBSKILL_CHECK')
```

Comment:
I'll have to check, but I'm not sure you need to remove listeners on death

Reply:
they get wiped out during the next spawn when the entity gets reconstructed at it init state again. same with localvar.